src/healthcheck_httpd.erl

```diff
@@ -77,22 +77,43 @@ init(Req, Opts) ->
 %%% private
 %%%===================================================================
 
-read_calver(Path) ->
+read_semver_file(Path, Number) ->
     case file:read_file(Path) of
-            {ok, Data} -> lists:nth(1, binary:split(Data, [<<"\n">>], [trim_all]));
+            {ok, Data} -> lists:nth(Number, binary:split(Data, [<<"\n">>], [global]));
             {error, _} -> <<"file error">>
     end.
 
-find_calver(Path) ->
-    Data = read_calver(Path),
+find_version_number(Path) ->
+    Data = read_semver_file(Path, 1),
     case re:run(binary_to_list(Data),
-                <<"(\\d{4}).(\\d{2}).(\\d{2}).(\\S{0,10}|$)">>,
+                <<"(\\d{1,}).(\\d{1,}).(\\d{1,})-(\\S{0,}|$)">>,
                 [{capture, all_but_first, list}]) of
-        {match, [Year, Mounth, Day, Description]} ->
-                Year ++ "." ++ Mounth ++ "." ++ Day ++ "." ++ Description;
+        {match, [Major, Minor, Patch, Label]} ->
+                Major ++ "." ++ Minor ++ "." ++ Patch ++ "-" ++ Label;
         nomatch -> "undefined"
     end.
 
+find_version_datetime(Path) ->
+    Data = read_semver_file(Path, 2),
+    case re:run(binary_to_list(Data),
+                <<"(\\d{1,2})/(\\d{1,2}).(\\d{2,4}) (\\d{2}):(\\d{2}):(\\d{2})">>,
+                [{capture, all_but_first, list}]) of
+        {match, [Day, Month, Year, Hours, Minutes, Seconds]} ->
+                Day ++ "/" ++ Month ++ "/" ++ Year ++ " " ++ Hours ++ ":" Minutes ++ ":" Seconds;
+        nomatch -> "undefined"
+    end.
+
+
+find_service_name(Path) ->
```

Comment:
Do we need this validation here? SEMVER file has been generated automatically, I bet it's not gonna be modified by hand, probably just get 3rd line from SEMVER and that's it?

Reply:
fixed